python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -300,26 +304,27 @@ def fn(item: Any) -> Any:
                         **fn_kwargs,
                     )
                 except Exception as e:
-                    _handle_debugger_exception(e)
+                    _handle_debugger_exception(e, item)
 
     else:
 
         def fn(item: Any) -> Any:
             try:
                 return op_fn(item, *fn_args, **fn_kwargs)
             except Exception as e:
-                _handle_debugger_exception(e)
+                _handle_debugger_exception(e, item)
 
         def init_fn():
             pass
 
     return fn, init_fn
 
 
-def _handle_debugger_exception(e: Exception):
+def _handle_debugger_exception(e: Exception, item: Any = None):
     """If the Ray Debugger is enabled, keep the full stack trace unmodified
     so that the debugger can stop at the initial unhandled exception.
     Otherwise, clear the stack trace to omit noisy internal code path."""
+    logger.error(f"Failed to process the following data block: {str(item)}")
```

Comment:
* not sure if `logger.error` can correctly capture the exception in this case. let's add a unit test? 
* `str()` is redundant.

Reply:
actually, I think we can just wrap this error message in `UserCodeException`. So it'd be printed from the driver. 
for the other if branch, we can use `logger.error`.